### PR TITLE
Binary chatsounds list format

### PR DESCRIPTION
**Pull request by PotcFdk**
This pull request introduces a new list format.

The list loader for `lists_nosend` has been modified to support the CSND format (Chatsounds List Format Specification Version 1, draft 1 (see below)).
# Comparison:
### Binary loader

# Specification draft

[csnd-1draft1.txt](https://github.com/Metastruct/garrysmod-chatsounds/files/505047/csnd-1draft1.txt)
# Test data

Example data for all the currently existing sound sets can be evaluated using the [binary-lists-test-data-draft1](https://github.com/Metastruct/garrysmod-chatsounds/tree/binary-lists-test-data-draft1) branch. That branch also contains the `cl_core.lua` from this pull request, so it should be fully functional (at least in single-player; interplay with the multiplayer mode (incuding the server having its own copy) has not been tested and is irrelevant for this matter).
Note, that the files still use identical paths compared to upstream. This is to keep the changes in `cl_core.lua` at a minimum and it should be changed before putting into production.
# Pull Request

Use this pull request to comment on whether the change of the list format is a good idea. Please consider:
- Performance
- Practicality
- ?

### CODE REVIEW

**Comment by KabusIblis** [comment]
only one approver with nobody testing isnt a valid enough reason to push something that was last touched 6 months ago

**Comment by Noiwex** [comment]
Shut up nigger.

**Comment by PotcFdk** [comment]
@KabusIblis It doesn't matter how long it has been. It doesn't say anything about how well the format would work for us.
@Noiwex Please don't do that. Also this should require more dev-approvals than just one.

**Comment by davue** [comment]
Would also make the loading of the sound list easier and more consistent in non-lua environments.

**Comment by ajloveslily14** [comment]
"The first 4 bytes specify the sound file duration in milliseconds.
      If this is zero, the sound file shall be skipped."

What about sounds that loop infinitely?

**Comment by Noiwex** [comment]
Looping sound shall have its own flag.

**Comment by ajloveslily14** [comment]
It would require a rewrite of the generator I would imagine. Other then that I think that aslong as it doesn't change file paths to sounds or doesn't change chatsounds.SortedList(much anyways) it seems good to me. Likely help with file sizes of lists too, though that's not as big of an issue. All we'd need next is an efficient way to compress and decompress sounds so the repo isn't so big, lol.

**Comment by Earu** [comment]
@PotcFdk is this issue still relevant, considering we use @CapsAdmin chatsounds now?

**Comment by PotcFdk** [comment]
AFAIK there is no canon as to how to proceed with Chatsounds and my vision of a modularized repository system and a central standards committee failed because I didn't have enough time to form it and nobody else did.
Introducing the changes from this PR doesn't make a big enough impact to invest the time into somehow making it work again with different systems, seeing as it would just be "fixing" instead of being part of a clean redesign. 
In other words, it made sense as a list format draft for a standardisation that never took place.
